packages/expo-facebook/android/src/main/java/expo/modules/facebook/FacebookModule.kt

```diff
@@ -0,0 +1,342 @@
+  }
+
+  private fun accessTokenToBundle(accessToken: AccessToken?): Bundle? {
+    return if (accessToken != null) {
+      Bundle().apply {
+        putString("token", accessToken.token)
+        putString("userId", accessToken.userId)
+        putString("appId", accessToken.applicationId)
+        putStringArrayList("permissions", ArrayList(accessToken.permissions))
+        putStringArrayList("declinedPermissions", ArrayList(accessToken.declinedPermissions))
+        putStringArrayList("expiredPermissions", ArrayList(accessToken.expiredPermissions))
+        putDouble("expirationDate", accessToken.expires.time.toDouble())
+        putDouble("dataAccessExpirationDate", accessToken.dataAccessExpirationTime.time.toDouble())
+        putDouble("refreshDate", accessToken.lastRefresh.time.toDouble())
+        putString("tokenSource", accessToken.source.name)
+      }
+    } else {
+      null
+    }
+  }
+
+  @ExpoMethod
+  fun initializeAsync(options: ReadableArguments, promise: Promise) {
+    try {
+      options.getString("appId")?.let {
+        appId = it
+        FacebookSdk.setApplicationId(it)
+      }
+      if (options.containsKey("appName")) {
+        appName = options.getString("appName")
+        FacebookSdk.setApplicationName(appName)
+      }
+      if (options.containsKey("version")) {
+        FacebookSdk.setGraphApiVersion(options.getString("version"))
+      }
+      if (options.containsKey("autoLogAppEvents")) {
+        val autoLogAppEvents = options.getBoolean("autoLogAppEvents")
+        FacebookSdk.setAutoLogAppEventsEnabled(autoLogAppEvents)
+      }
+      if (options.containsKey("domain")) {
+        FacebookSdk.setFacebookDomain(options.getString("domain"))
+      }
+      if (options.containsKey("isDebugEnabled")) {
+        FacebookSdk.setIsDebugEnabled(options.getBoolean("isDebugEnabled"))
+      }
+      FacebookSdk.sdkInitialize(context) {
+        FacebookSdk.fullyInitialize()
+        appId = FacebookSdk.getApplicationId()
+        appName = FacebookSdk.getApplicationName()
+        appEventLogger = AppEventsLogger.newLogger(context)
+        attributionIdentifiers = AttributionIdentifiers.getAttributionIdentifiers(context)
+        promise.resolve(null)
+      }
+    } catch (e: Exception) {
+      promise.reject(ERR_FACEBOOK_MISCONFIGURED, "An error occurred while trying to initialize a FBSDK app", e)
+    }
+  }
+
+  @ExpoMethod
+  fun logOutAsync(promise: Promise) {
+    AccessToken.setCurrentAccessToken(null)
+    LoginManager.getInstance().logOut()
+    promise.resolve(null)
+  }
+
+  @ExpoMethod
+  fun logInWithReadPermissionsAsync(config: ReadableArguments, promise: Promise) {
+    if (FacebookSdk.getApplicationId() == null) {
+      promise.reject(
+        ERR_FACEBOOK_MISCONFIGURED,
+        "No appId configured, required for initialization. " +
+          "Please ensure that you're either providing `appId` to `initializeAsync` as an argument or inside AndroidManifest.xml."
+      )
+    }
```

Comment:
Not directly related to this PR, but shouldn't the method return here as the promise is rejected anyway? I see the `return` was missing in Java too. 🤔